apis/python/src/tiledbsoma/soma_dataframe.py

```diff
@@ -285,22 +262,17 @@ def write(self, values: pa.Table) -> None:
                     )
                 )
 
-        if self._get_is_sparse():
-            # sparse write
-            with self._tiledb_open("w") as A:
-                A[rowids] = attr_cols_map
-        else:
-            # TODO: This was a quick thing to bootstrap some early ingestion tests but needs more thought.
-            # In particular, rowids needn't be either zero-up or contiguous.
-            assert len(rowids) > 0
-            rowids = sorted(rowids)
-            assert rowids[0] == 0
-
-            # dense write
-            lo = rowids[0]
-            hi = rowids[-1]
-            with self._tiledb_open("w") as A:
-                A[lo : (hi + 1)] = attr_cols_map
+        # TODO: This was a quick thing to bootstrap some early ingestion tests but needs more thought.
+        # In particular, rowids needn't be either zero-up or contiguous.
+        assert len(rowids) > 0
+        rowids = sorted(rowids)
+        assert rowids[0] == 0
+
+        # dense write
+        lo = rowids[0]
```

Comment:
This is not what the spec says - it allows for a scattered write (ie, each row in the RecordBatch can have a non-contiguous soma_rowid).

I don't think TileDB Dense arrays support this scattered indexing, for either read or write. We ran into the same issue in DenseNdArray, and changed the read/write operations to require coordinates be contiguous (ie, they were specified separately, as either a single `int` or as a slice).

We may need to do the same for DataFrame.